pkg/controller/stacks/stack/stack.go

```diff
@@ -670,30 +670,53 @@ func (h *stackHandler) prepareHostAwareJob(j *batch.Job, tokenSecret string) err
 	return nil
 }
 
-func (h *stackHandler) processDeployment(ctx context.Context) error {
+func (h *stackHandler) prepareDeployment() *apps.Deployment {
 	controllerDeployment := h.ext.Spec.Controller.Deployment
 	if controllerDeployment == nil {
 		return nil
 	}
 
-	// ensure the deployment is set to use this stack's service account that we created
+	// force the deployment to use stack opinionated names and service account
+	name := h.ext.Name + "-controller"
+	matchLabels := map[string]string{"app": name}
```

Comment:
What are we using this [selector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#deploymentspec-v1-apps) for?

Reply:
The selector is required by the Deployment spec and without it the `StackDefinition` resource  (yes, that) is not valid and is rejected on create.

Reply:
I think I was answering the wrong question here.  I needed to include a selector in the emitted `StackDefinition` resource.

And later when the Stack controller finally assembles the deployment, the deployment must have a selector to identify the pods that it is responsible for.